### DISCUSSION TITLE
Unable to test actual use cases from my application in Sandbox mode

### DISCUSSION THREAD

**Discussion by ivancortesromero** [amzn/selling-partner-api-models#4311] [Q&A] [answered]
I have an app that receives orders from Amazon and fulfills them. An important process step is executing packaging rules to optimize the space inside the boxes depending on the sizes of the items in the orders. Testing this use case is impossible with the sandbox since the responses for Orders API are statically mocked. I’ve considered testing it on production with real products by setting a high price so that actual buyers don't buy my products yet but I’m afraid Amazon might ban me as this doesn’t seem like a typical or acceptable practice. I assume someone else has encountered this issue, and I'd like to know how you have solved it.

**Comment by chrisskhoury** [comment, marked as answer]
Hello @ivancortesromero,

Testing the packaging rules for order fulfillment can indeed be challenging with the limitations of the sandbox environment. While the sandbox is useful for understanding API structure and payloads, it does not fully support realistic testing scenarios.

Here are a few approaches you can consider to test your use case without violating Amazon's policies. You can use real products but control the testing conditions carefully:

- **List Low-Demand Products:** Use products that are unlikely to sell quickly, even if priced reasonably. Some developers use company merchandise, such as branded caps or stickers, as test products.
- **Setup a Low-Priced Product:** Set the product price very low so that you can place the test order yourself.
- **Set a Test Window:** Perform your tests during a specific window of time, then remove or adjust the listings promptly after testing.
- **Cancel the Order After Testing:** Ensure to cancel the fulfillment after testing the flow.
- **Communicate with Amazon:** Inform Amazon Seller Support about your testing activities to ensure they understand your intent and avoid any misunderstandings or potential bans.

I hope these suggestions help you find a suitable solution. If you have any further questions or need additional assistance, feel free to ask.

Best regards,  

Chris  
Solution Architect

**Reply by ivancortesromero** [reply]
Thank you, Chris, for your answer. Does Amazon plan to enable a sandbox where developers can replicate real dynamics as in production?
